### PR TITLE
feat(rome_css_parser): CSS lexer #4682

### PR DESCRIPTION
**Pull request by denbezrukov**
## Summary

Lexes string literal and comments 

https://drafts.csswg.org/css-syntax/#consume-string-token
https://github.com/swc-project/swc/blob/main/crates/swc_css_parser/src/lexer/mod.rs
https://github.com/servo/rust-cssparser/blob/master/src/tokenizer.rs

## Test Plan

`cargo test -p rome_css_parser`

### CODE REVIEW

File: crates/rome_css_parser/src/lexer/mod.rs [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+                                    .and_then(|c| (c as char).to_digit(16)) else { break; };
+                                self.advance(1);
+
+                                hex = hex * 16 + digit;
+                            }
+
+                            // Interpret the hex digits as a hexadecimal number. If this number is zero, or
+                            // is for a surrogate, or is greater than the maximum allowed code point, return
+                            // U+FFFD REPLACEMENT CHARACTER (�).
+                            let hex = match hex {
```
Comment (denbezrukov) [inline crates/rome_css_parser/src/lexer/mod.rs:343]: Here, we need to convert the escaped sequence to a char. swc uses two buffers for this: raw and buf.
https://github.com/swc-project/swc/blob/e4f9f734ad1f92c6f05e8cb4c2d799679cca9f79/crates/swc_css_parser/src/lexer/mod.rs#L692-L723

I'm not sure if we can implement this with our CST, but we can potentially implement such logic at the syntax level.

File: crates/rome_css_parser/src/lexer/mod.rs [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+
+                self.diagnostics.push(err);
+
+                if has_newline {
+                    MULTILINE_COMMENT
+                } else {
+                    COMMENT
+                }
+            }
+            Some(b'/') => {
```
Comment (denbezrukov) [inline crates/rome_css_parser/src/lexer/mod.rs:447]: `//` comments are forbidden in CSS, but I prefer to keep it.
 We can implement an option to turn on/off such comments.

File: crates/rome_css_parser/src/lexer/mod.rs (hunk 1 above)
Comment (ematipico) [reply crates/rome_css_parser/src/lexer/mod.rs:343]: True, in CSS we need two values 🤔 we can think about it later, as you suggested

**Review by ematipico — APPROVED** [review]
A great start! Thank you @denbezrukov

**Comment by DeadlyMissile** [comment]
Is there a reason why you are creating a css lexer from scratch? Why not use one that exists already? This feels like ALOT of work

**Comment by ematipico** [comment]
> Is there a reason why you are creating a css lexer from scratch? Why not use one that exists already? This feels like ALOT of work

The lexers that already exists are not suitable for CSTs, and tree-sitter APIs are very poor

**Pull request merged**